src/main/java/com/synopsys/integration/detect/lifecycle/run/RunManager.java

```diff
@@ -401,17 +414,17 @@ private void runBlackDuckProduct(final ProductRunData productRunData, final Dete
         logger.info(ReportConstants.RUN_SEPARATOR);
         if (null != blackDuckServicesFactory) {
             logger.info("Will perform Black Duck post actions.");
-            final BlackDuckPostOptions blackDuckPostOptions = detectConfigurationFactory.createBlackDuckPostOptions();
-            final BlackDuckPostActions blackDuckPostActions = new BlackDuckPostActions(blackDuckServicesFactory, eventSystem);
+            BlackDuckPostOptions blackDuckPostOptions = detectConfigurationFactory.createBlackDuckPostOptions();
+            BlackDuckPostActions blackDuckPostActions = new BlackDuckPostActions(blackDuckServicesFactory, eventSystem);
             blackDuckPostActions.perform(blackDuckPostOptions, codeLocationWaitController, projectVersionWrapper, detectConfigurationFactory.findTimeoutInSeconds());
 
-            if ((bdioResult.getUploadTargets().size() > 0 || detectToolFilter.shouldInclude(DetectTool.SIGNATURE_SCAN))) {
-                final Optional<String> componentsLink = Optional.ofNullable(projectVersionWrapper)
-                                                            .map(ProjectVersionWrapper::getProjectVersionView)
-                                                            .flatMap(projectVersionView -> projectVersionView.getFirstLink(ProjectVersionView.COMPONENTS_LINK));
+            if ((!bdioResult.getUploadTargets().isEmpty() || detectToolFilter.shouldInclude(DetectTool.SIGNATURE_SCAN))) {
+                Optional<String> componentsLink = Optional.ofNullable(projectVersionWrapper)
+                                                      .map(ProjectVersionWrapper::getProjectVersionView)
+                                                      .flatMap(projectVersionView -> projectVersionView.getFirstLink(ProjectVersionView.COMPONENTS_LINK));
 
                 if (componentsLink.isPresent()) {
-                    final DetectResult detectResult = new BlackDuckBomDetectResult(componentsLink.get());
+                    DetectResult detectResult = new BlackDuckBomDetectResult(componentsLink.get());
                     eventSystem.publishEvent(Event.ResultProduced, detectResult);
                 }
             }
```

Comment:
did these changes for runBlackDuckProduct reduce the complexity enough?

Reply:
Not the final removals. I did slightly restructure one method. There were multiple cognitive complexity issues in this file. I am only addressing one of them here. Not quite sure how I want to tackle this file going forward.